### PR TITLE
Fix SB extension tests

### LINKED ISSUE #27353: [Servicebus] MessageDraining_QueueWithSessions_Batch failing in nightly runs
Servicebus nightly test runs are failing with:

> Error message
> Microsoft.Azure.WebJobs.Host.Indexers.FunctionIndexingException : Error indexing method 'ServiceBusMultipleMessagesTestJob_BindToMessageArray.Run'
> ----> System.InvalidOperationException : Cannot bind parameter 'lockedUntilArray' to type DateTimeOffset[]. Make sure the parameter Type is supported by the binding. The binding supports the following parameter names for type DateTimeOffset[]: (ExpiresAtArray, EnqueuedTimeArray). If you're trying to bind to one of those values, rename your parameter to match the contract name (case insensitive). If you're using binding extensions (e.g. Azure Storage, ServiceBus, Timers, etc.) make sure you've called the registration method for the extension(s) in your startup code (e.g. builder.AddAzureStorage(), builder.AddServiceBus(), builder.AddTimers(), etc.).
> 
> 
> Stack trace
>    at Microsoft.Azure.WebJobs.Host.RecoverableException.TryRecover(ILogger logger) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Exceptions\RecoverableException.cs:line 81
>    at Microsoft.Azure.WebJobs.Host.Indexers.FunctionIndexer.IndexTypeAsync(Type type, IFunctionIndexCollector index, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Indexers\FunctionIndexer.cs:line 94
>    at Microsoft.Azure.WebJobs.Host.Indexers.FunctionIndexProvider.CreateAsync(CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Indexers\FunctionIndexProvider.cs:line 97
>    at Microsoft.Azure.WebJobs.Host.Indexers.FunctionIndexProvider.GetAsync(CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Indexers\FunctionIndexProvider.cs:line 69
>    at Microsoft.Azure.WebJobs.Host.Executors.JobHostContextFactory.Create(JobHost host, CancellationToken shutdownToken, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\JobHostContextFactory.cs:line 119
>    at Microsoft.Azure.WebJobs.JobHost.InitializeHostAsync(CancellationToken cancellationToken, TaskCompletionSource`1 initializationTask) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\JobHost.cs:line 347
>    at Microsoft.Azure.WebJobs.JobHost.StartAsyncCore(CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\JobHost.cs:line 95
>    at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
>    at Microsoft.Azure.WebJobs.Host.EndToEndTests.WebJobsServiceBusTestBase.BuildHost[TJobClass](Action`1 configurationDelegate, Boolean startHost, Boolean useTokenCredential) in /Users/runner/work/1/s/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/WebJobsServiceBusTestBase.cs:line 174
>    at Microsoft.Azure.WebJobs.Host.EndToEndTests.ServiceBusEndToEndTests.TestMultiple[T](Boolean isXml, Action`1 configurationDelegate) in /Users/runner/work/1/s/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/ServiceBusEndToEndTests.cs:line 615
>    at Microsoft.Azure.WebJobs.Host.EndToEndTests.ServiceBusEndToEndTests.TestBatch_Messages() in /Users/runner/work/1/s/sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/ServiceBusEndToEndTests.cs:line 191
>    at NUnit.Framework.Internal.TaskAwaitAdapter.GenericAdapter`1.BlockUntilCompleted()
>    at NUnit.Framework.Internal.MessagePumpStrategy.NoMessagePumpStrategy.WaitForCompletion(AwaitAdapter awaiter)
>    at NUnit.Framework.Internal.AsyncToSyncAdapter.Await(Func`1 invoke)
>    at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod(TestExecutionContext context)
>    at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
>    at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.<>c__DisplayClass1_0.<Execute>b__0()
>    at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.RunTestMethodInThreadAbortSafeZone(TestExecutionContext context, Action action)
> --InvalidOperationException
>    at Microsoft.Azure.WebJobs.Host.Indexers.FunctionIndexer.IndexMethodAsyncCore(MethodInfo method, IFunctionIndexCollector index, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Indexers\FunctionIndexer.cs:line 273
>    at Microsoft.Azure.WebJobs.Host.Indexers.FunctionIndexer.IndexMethodAsync(MethodInfo method, IFunctionIndexCollector index, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Indexers\FunctionIndexer.cs:line 159

For more details check here:

- https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1406774&view=results

@jsquire for notification.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/27353

### CODE REVIEW

File: sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/ServiceBusEndToEndTests.cs [hunk 1]
```diff
@@ -1070,7 +1070,6 @@ public class ServiceBusMultipleMessagesTestJob_BindToMessageArray
                 DateTimeOffset[] expiresAtArray,
                 DateTime[] enqueuedTimeUtcArray,
                 DateTimeOffset[] enqueuedTimeArray,
-                DateTimeOffset[] lockedUntilArray,
```
Comment (JoshLove-msft) [inline sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/ServiceBusEndToEndTests.cs:1073]: This was removed in a later commit of https://github.com/Azure/azure-sdk-for-net/pull/27308

**Review by jsquire — APPROVED** [review]

**Pull request merged**